pkg/controller/account/iam_test.go

```diff
@@ -0,0 +1,367 @@
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubectl/pkg/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+type mocks struct {
+	fakeKubeClient client.Client
+	mockCtrl       *gomock.Controller
+}
+
+// setupDefaultMocks is an easy way to setup all of the default mocks
+func setupDefaultMocks(t *testing.T, localObjects []runtime.Object) *mocks {
+	mocks := &mocks{
+		fakeKubeClient: fake.NewFakeClient(localObjects...),
+		mockCtrl:       gomock.NewController(t),
+	}
+
+	return mocks
+}
+
+func TestIAMCreateSecret(t *testing.T) {
+
+	err := apis.AddToScheme(scheme.Scheme)
+	if err != nil {
+		fmt.Printf("failed adding to scheme in iam_test.go")
+	}
+
+	secret := CreateSecret(
+		"test",
+		"namespace",
+		map[string][]byte{
+			"one": []byte("hello"),
+			"two": []byte("world"),
+		},
+	)
+	mocks := setupDefaultMocks(t, []runtime.Object{})
+
+	// This is necessary for the mocks to report failures like methods not being called an expected number of times.
+	// after mocks is defined
+	defer mocks.mockCtrl.Finish()
+
+	r := ReconcileAccount{
+		Client: mocks.fakeKubeClient,
+		scheme: scheme.Scheme,
+	}
+
+	nullLogger := testutils.NullLogger{}
+	account := newTestAccountBuilder().acct
+	err = r.CreateSecret(nullLogger, &account, secret)
+	assert.Nil(t, err)
+
+	/* Can't run a second time ->
+
+	err = r.CreateSecret(nullLogger, &account, secret)
+	assert.Nil(t, err)
+
+
+	Expected nil, but got: &runtime.notRegisteredErr{
+		schemeName:"pkg/runtime/scheme.go:101",
+		gvk:schema.GroupVersionKind{Group:"", Version:"", Kind:""},
+		target:runtime.GroupVersioner(nil), t:(*reflect.rtype)(0x148ad20)}
+	*/
+}
+
+func TestGetSTSCredentials(t *testing.T) {
+
+	mocks := setupDefaultMocks(t, []runtime.Object{})
+	nullLogger := testutils.NullLogger{}
+	mockAWSClient := mock.NewMockClient(mocks.mockCtrl)
+
+	AccessKeyId := aws.String("MyAccessKeyID")
+	Expiration := aws.Time(time.Now().Add(time.Hour))
+	SecretAccessKey := aws.String("MySecretAccessKey")
+	SessionToken := aws.String("MySessionToken")
+
+	mockAWSClient.EXPECT().AssumeRole(gomock.Any()).Return(
+		&sts.AssumeRoleOutput{
+			Credentials: &sts.Credentials{
+				AccessKeyId:     AccessKeyId,
+				Expiration:      Expiration,
+				SecretAccessKey: SecretAccessKey,
+				SessionToken:    SessionToken,
+			},
+		},
+		nil, // no error
+	)
+
+	creds, err := getSTSCredentials(
+		nullLogger,
+		mockAWSClient,
+		"",
+		"",
+		"",
+	)
+
+	assert.Equal(t, creds.Credentials.AccessKeyId, AccessKeyId)
+	assert.Equal(t, creds.Credentials.Expiration, Expiration)
+	assert.Equal(t, creds.Credentials.SecretAccessKey, SecretAccessKey)
+	assert.Equal(t, creds.Credentials.SessionToken, SessionToken)
+	assert.NoError(t, err)
+
+	// TODO Test AWS Failure
+}
+
+func TestRetryIfAwsServiceFailureOrInvalidToken(t *testing.T) {
```

Comment:
This one could be converted to a table-driven test (https://github.com/golang/go/wiki/TableDrivenTests)